mllib-local/src/vectorized/java/org/apache/spark/ml/linalg/VectorizedBLAS.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.linalg;
+
+import com.github.fommil.netlib.F2jBLAS;
+import jdk.incubator.vector.DoubleVector;
+import jdk.incubator.vector.FloatVector;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.VectorSpecies;
+
+public class VectorizedBLAS extends F2jBLAS {
+
+  private final static VectorSpecies<Float>  FMAX = FloatVector.SPECIES_MAX;
+  private final static VectorSpecies<Double> DMAX = DoubleVector.SPECIES_MAX;
+
+  // y += alpha * x
+  @Override
+  public void daxpy(int n, double alpha, double[] x, int incx, double[] y, int incy) {
+    if (incx == 1 && incy == 1 && n <= x.length && n <= y.length) {
+      if (alpha != 0.) {
```

Comment:
Can we move this line before line 34?

Reply:
Same as https://github.com/apache/spark/pull/30810#discussion_r546452297